File: problems/p1848_minimum_distance_to_the_target_element/test_solution.cpp
#include "solution.cpp"
#include <cassert>
#include <iostream>

int main() {
    Solution s;

    // Example 1
    assert(s.getMinDistance({1,2,3,4,5}, 5, 3) == 1);
    std::cout << "test_example_1 PASSED\n";

    // Example 2
    assert(s.getMinDistance({1}, 1, 0) == 0);
    std::cout << "test_example_2 PASSED\n";

    // Example 3
    assert(s.getMinDistance({1,1,1,1,1,1,1,1,1,1}, 1, 0) == 0);
    std::cout << "test_example_3 PASSED\n";

    std::cout << "\nAll tests passed!\n";
    return 0;
}

File: problems/p1391_check_if_there_is_a_valid_path_in_a_grid/test_solution.py
from .solution import Solution


def test_example_1():
    assert Solution().hasValidPath(grid=[[2,4,3],[6,5,2]]) == True


def test_example2():
    assert Solution().hasValidPath(grid=[[1,2,1],[1,2,1]]) == False



def test_example_3():
    assert Solution().hasValidPath(grid=[[1,1,2]]) == False
